client/extensions/woocommerce/state/sites/settings/stripe-connect-account/test/reducer.js

```diff
@@ -343,4 +347,176 @@ describe( 'reducer', () => {
+								isDisconnecting: false,
+							},
+						},
+					},
+				},
+				action
+			);
+			expect( newState[ 123 ].settings.stripeConnectAccount.isDisconnecting ).to.eql( true );
+			expect( newState[ 456 ].settings.stripeConnectAccount.isDisconnecting ).to.eql( false );
+		} );
+	} );
+
+	// TODO
+	describe( 'connectAccountDisconnectComplete Success', () => {
+		test( 'should update state', () => {
+			const action = {
+				type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_DISCONNECT_COMPLETE,
+				siteId: 123,
+			};
+			const newState = stripeConnectAccountReducer( undefined, action );
+			expect( newState ).to.eql( {
+				connectedUserID: '',
+				displayName: '',
+				email: '',
+				error: '',
+				firstName: '',
+				isActivated: false,
+				isDisconnecting: false,
+				isRequesting: false,
+				lastName: '',
+				logo: '',
+			} );
+		} );
+
+		test( 'should leave other sites state unchanged', () => {
+			const action = {
+				type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_DISCONNECT_COMPLETE,
+				siteId: 123,
+			};
+			const newState = sitesReducer(
+				{
+					123: {
+						settings: {
+							stripeConnectAccount: {
+								connectedUserID: 'acct_25rzu7Alijdnw0FB',
+								displayName: 'Bar Foo',
+								email: 'bar@foo.com',
+								error: '',
+								firstName: 'Bar',
+								isActivated: false,
+								isDisconnecting: true,
+								isRequesting: false,
+								lastName: 'Foo',
+								logo: '',
+							},
+						},
+					},
+					456: {
+						settings: {
+							stripeConnectAccount: {
+								connectedUserID: 'acct_14qyt6Alijdnw0EA',
+								displayName: 'Foo Bar',
+								email: 'foo@bar.com',
+								error: '',
+								firstName: 'Foo',
+								isActivated: false,
+								isDisconnecting: false,
+								isRequesting: false,
+								lastName: 'Bar',
+								logo: '',
+							},
+						},
+					},
+				},
+				action
+			);
+			expect( newState[ 123 ].settings.stripeConnectAccount.isRequesting ).to.eql( false );
+			expect( newState[ 123 ].settings.stripeConnectAccount.connectedUserID ).to.eql( '' );
+			expect( newState[ 456 ].settings.stripeConnectAccount.connectedUserID ).to.eql(
+				'acct_14qyt6Alijdnw0EA'
+			);
+		} );
+	} );
+
+	// TODO
+	describe( 'connectAccountDisconnectComplete w/ Error', () => {
+		test( 'should set the error in state', () => {
+			const action = {
+				type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_DISCONNECT_COMPLETE,
+				siteId: 123,
+				error: 'My error',
+			};
+			const newState = stripeConnectAccountReducer( undefined, action );
+			expect( newState.error ).to.eql( 'My error' );
+		} );
+
+		test( 'should leave other sites state unchanged', () => {
+			const action = {
+				type: WOOCOMMERCE_SETTINGS_STRIPE_CONNECT_ACCOUNT_DISCONNECT_COMPLETE,
+				siteId: 123,
+				error: 'My error',
+			};
+			const newState = sitesReducer(
+				{
+					123: {
+						settings: {
+							stripeConnectAccount: {
+								connectedUserID: 'acct_14qyt6Alijdnw0EA',
```

Comment:
same {} as below - maybe extract

Reply:
I'd like to leave it as is for now and consider creating some common fixtures after all the flows are hooked up.